yt_dlp/extractor/reddit.py

```diff
@@ -158,6 +182,37 @@ def add_thumbnail(src):
                 'duration': int_or_none(reddit_video.get('duration')),
             }
 
+        # Check if media is hosted on reddit as an image (but also available as video):
+        reddit_video = traverse_obj(data, ('preview', 'images', 0, 'variants'))
+        if isinstance(reddit_video, dict):
```

Comment:
```suggestion
        reddit_video = traverse_obj(data, ('preview', 'images', 0, 'variants'), expected_type=dict)
        if reddit_video:
```
imo, the expected_type is not usefull in practice, but 🤷